### ISSUE TITLE
Our work is focused on structural generalization and uses the provided Klinger et al 2024 vocab description of COGS - Mention 2 out of distribution words not in main training file (in train_100.tsv and dev.tsv, not leaked from test/gen) from Klinger vocab affecting a small number of gen examples (COGS author says ignore)

### ISSUE THREAD

**Issue by willy-b** [willy-b/learning-rasp#20]
re: in najoungkim / COGS/issues/2#issuecomment-976216841 (see link from https://github.com/willy-b/RASP-for-ReCOGS/issues/33 back to COGS author repo, don't want to spam them by mentioning too many places)

"gardner" and "monastery" are not represented in the main COGS/ReCOGS training data but are present in a handful of generalization/test examples (only such nouns where that occurs, I am checking coverage of verbs too now).

We use the COGS vocabulary provided by Klinger et al 2024 (IBM) ( https://github.com/IBM/cpg/blob/c3626b4e03bfc681be2c2a5b23da0b48abe6f570/src/model/cogs_data.py#L523 ) (we do not use their model just their description of the COGS input grammar)
in building our RASP model which operates on part-of-speech/verb type tokens and as an "embedding" maps the words using the Klinger dictionary to part-of-speech/verb type before working from there.

Therefore we inherit the "gardner" and "monastery" in our vocabulary from Klinger et al 2024 via https://github.com/IBM/cpg/blob/c3626b4e03bfc681be2c2a5b23da0b48abe6f570/src/model/cogs_data.py#L389 and https://github.com/IBM/cpg/blob/c3626b4e03bfc681be2c2a5b23da0b48abe6f570/src/model/cogs_data.py#L377 .

Should we be excluding those?

(one could exclude and re-run since this is not a neural network model but a series of RASP rules so one can update the map and look at exactly how it behaves - though it will just not process those lines which will not qualitatively affect the result)

Kim and Linzen 2020 actually answer this question in their repository and say that these were out of distribution was unintentional and to exclude them when doing evaluations:
najoungkim / COGS/issues/2#issuecomment-976216841 (see link from https://github.com/willy-b/RASP-for-ReCOGS/issues/33 back to COGS author repo, don't want to spam them by mentioning too many places)

So this issue is to track adding a note to our code here that we are not testing generalization to unseen words which occurs with these two words so we include them in our vocabulary instead of excluding the examples so we can still test whether the prepositional phrase recursion or other features being tested in those examples works.

**Comment by willy-b** [comment]
This potential cause for concern (which is not actually a cause for concern) is now explicitly discussed in the paper so going to close this ticket here. For comparison in doing the separate https://github.com/willy-b/RASP-for-COGS I decided to exclude the 2 words only mentioned in the additional training files and dev.tsv (not leaked from gen/test, just not in the main train.tsv file) and support out of vocabulary words there (tokenized as unknown token) and there was negligible effect on performance (none except intentionally missing the examples with "monastery" and "gardner" which leads to missing 1 out of 3000 in the test set and a very small number of generalization examples featuring those words). 

Separately, there were also unreachable stemmed verbs in the decoder vocab (not encoder, not reachable, and not in gen/test file so not affecting the evaluations) built with reference to Klinger verb_lemmas (not in encoder vocab so cannot be used to read OOV words, and they don't occur in the eval files anyway). We could remove from the decoder vocab the unreachable verbs: "snort", "envied", "cool", "melted", "torn", "snorted", "washed", "wash", "melt", "soaked", "tore", "expanded", "tear", "envy", "bought", "expand", "soak", "buy", "cooled", "hit", "popped", "pop" ("crawled" only is retained as crawl is present in the training data as a primitive though may remove that also). 

Those are not present in ReCOGS test.tsv/gen.tsv and even if had been also in encoder vocab would not have any effect on test/gen performance. Since they would not effect evaluations and I prefer not to make edits to this, I did not remove these, though I excluded them in the separate RASP-for-COGS project.
